app_flutter/test/service/appengine_cocoon_test.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:app_flutter/service/appengine_cocoon.dart';
+import 'package:cocoon_service/protos.dart'
+    show Commit, CommitStatus, Stage, Task;
+import 'package:test/test.dart';
+
+import 'package:http/http.dart' show Response;
+import 'package:http/testing.dart';
+import 'package:fixnum/fixnum.dart';
+
+void main() {
+  // This is based off data the Cocoon backend sends out from v1.
+  // It doesn't map directly to protos since the backend does
+  // not use protos yet.
+  final String jsonGetStatsResponse = """
+        {
+          "Statuses": [
+            {
+            "Checklist": {
+              "Key": "iamatestkey", 
+              "Checklist": {
+                "FlutterRepositoryPath": "flutter/cocoon", 
+                "CreateTimestamp": 123456789, 
+                "Commit": {
+                  "Sha": "ShaShankHash", 
+                  "Author": {
+                    "Login": "ShaSha", 
+                    "avatar_url": "https://flutter.dev"
+                    }
+                  }
+                }
+              }, 
+              "Stages": [
+                {
+                  "Name": "devicelab",
+                  "Status": "Succeeded",
+                  "Tasks": [
+                    {
+                      "Key": "taskKey1",
+                      "Task": {
+                        "Attempts": 1,
+                        "CreateTimestamp": 1569353940885,
+                        "EndTimestamp": 1569354700642,
+                        "Flaky": false,
+                        "Name": "complex_layout_semantics_perf",
+                        "Reason": "",
+                        "RequiredCapabilities": ["linux/android"],
+                        "ReservedForAgentID": "linux2",
+                        "StageName": "devicelab",
+                        "StartTimestamp": 1569354594672,
+                        "Status": "Succeeded",
+                        "TimeoutInMinutes": 0
+                      }
+                    }
+                  ]
+                }
+              ]
+            }
+          ], 
+          "AgentStatuses": []
+        }
+  """;
+
+  group('AppEngine CocoonService', () {
+    AppEngineCocoonService service;
+
+    setUp(() async {
+      service = AppEngineCocoonService();
+      service.client = MockClient((request) {
+        return Future<Response>.delayed(Duration(microseconds: 100),
```

Comment:
Don't add artificial delays to test scipts.You can make this return a future by making the method body `async`.

Reply:
This is the best thing I learned today :+1: